modules/redpanda/redpanda_test.go

```diff
@@ -179,6 +179,112 @@ func TestRedpandaWithAuthentication(t *testing.T) {
 	}
 }
 
+func TestRedpandaWithBootstrapUserAuthentication(t *testing.T) {
+	ctx := context.Background()
+	// redpandaCreateContainer {
+	ctr, err := redpanda.Run(ctx,
+		"docker.redpanda.com/redpandadata/redpanda:v23.3.3",
+		redpanda.WithEnableSASL(),
+		redpanda.WithEnableKafkaAuthorization(),
+		redpanda.WithEnableWasmTransform(),
+		redpanda.WithNewServiceAccount("superuser-1", "test"),
+		redpanda.WithNewServiceAccount("superuser-2", "test"),
+		redpanda.WithNewServiceAccount("no-superuser", "test"),
+		redpanda.WithSuperusers("superuser-1", "superuser-2"),
+		redpanda.WithEnableSchemaRegistryHTTPBasicAuth(),
+		redpanda.WithEnableAdminAPIAuthentication(),
+	)
+	testcontainers.CleanupContainer(t, ctr)
+	require.NoError(t, err)
+	// }
+
+	// kafkaSeedBroker {
+	seedBroker, err := ctr.KafkaSeedBroker(ctx)
+	// }
+	require.NoError(t, err)
+
+	// Test successful authentication & authorization with all created superusers
+	serviceAccounts := map[string]string{
+		"superuser-1": "test",
+		"superuser-2": "test",
+	}
+
+	for user, password := range serviceAccounts {
+		kafkaCl, err := kgo.NewClient(
+			kgo.SeedBrokers(seedBroker),
+			kgo.SASL(scram.Auth{
+				User: user,
+				Pass: password,
+			}.AsSha256Mechanism()),
+		)
+		require.NoError(t, err)
+
+		kafkaAdmCl := kadm.NewClient(kafkaCl)
+		_, err = kafkaAdmCl.CreateTopic(ctx, 1, 1, nil, fmt.Sprintf("test-%v", user))
+		require.NoError(t, err)
+		kafkaCl.Close()
```

Comment:
suggestion: invert the order to avoid failing the test and leaking the client

Reply:
I fixed the ordering throughout the tests.